rust/src/ffi.rs

```diff
@@ -878,3 +878,97 @@ pub unsafe extern "C" fn zcashlc_free_account_metadata_key(ptr: *mut AccountMeta
         drop(key);
     }
 }
+
+/// Information about a lightwalletd instance and the state of the blockchain.
+#[repr(C)]
+pub struct LightdInfo {
```

Comment:
Question: why are we making a C type for the purpose of pushing something that's already encoded as a protobuf over the FFI? Let's just pass the protobuf.

Reply:
That's a very good point, I completely forgot about that point and instead explicitly parsed the protobuf here and then reconstruct it on the Swift side. I will fix this 😅